src/Adapter/Product/Update/ProductShopUpdater.php

```diff
@@ -0,0 +1,302 @@
+        // (e.g copy only prices but not stock)
+        $fields = array_merge(
+            $this->getAssociationFields($sourceProduct),
+            $this->getPricesFields($sourceProduct),
+            $this->getStockFields($sourceProduct),
+            $this->getSEOFields($sourceProduct),
+            $this->getOptionsFields($sourceProduct),
+            $this->getCustomizationFields($sourceProduct),
+            $this->getDateFields($sourceProduct)
+        );
+
+        if ($this->productRepository->isAssociatedToShop($productId, $targetShopId)) {
+            $qb = $this->getUpdateQueryBuilder($fields, $productId, $targetShopId);
+        } else {
+            $qb = $this->getInsertQueryBuilder($fields, $productId, $targetShopId);
+        }
+
+        foreach ($fields as $column => $value) {
+            $qb->setParameter($column, $value);
+        }
+        $qb->execute();
+    }
+
+    /**
+     * @param array $fields
+     * @param ProductId $productId
+     * @param ShopId $targetShopId
+     *
+     * @return QueryBuilder
+     */
+    private function getInsertQueryBuilder(
+        array $fields,
+        ProductId $productId,
+        ShopId $targetShopId
+    ): QueryBuilder {
+        $fields['id_product'] = $productId->getValue();
+        $fields['id_shop'] = $targetShopId->getValue();
+
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->insert($this->dbPrefix . 'product_shop')
+            ->values($this->formatQueryValues($fields))
+            ->setParameter('id_product', $productId->getValue())
+            ->setParameter('id_shop', $targetShopId->getValue())
+        ;
+
+        return $qb;
+    }
+
+    /**
+     * @param array $fields
+     * @param ProductId $productId
+     * @param ShopId $targetShopId
+     *
+     * @return QueryBuilder
+     */
+    private function getUpdateQueryBuilder(
+        array $fields,
+        ProductId $productId,
+        ShopId $targetShopId
+    ): QueryBuilder {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->update($this->dbPrefix . 'product_shop')
+            ->where('id_shop = :id_shop')
+            ->where('id_product = :id_product')
+            ->setParameter('id_shop', $targetShopId->getValue())
+            ->setParameter('id_product', $productId->getValue())
+        ;
+
+        $queryValues = $this->formatQueryValues($fields);
+        foreach ($queryValues as $column => $value) {
+            $qb->set($column, $value);
+        }
+
+        return $qb;
+    }
+
+    /**
+     * @param Product $product
+     *
+     * @return array<string, int>
+     */
+    private function getAssociationFields(Product $product): array
+    {
+        return [
+            'id_category_default' => (int) $product->id_category_default,
+            'cache_default_attribute' => $product->cache_default_attribute,
```

Comment:
```suggestion
            'cache_default_attribute' => (int) $product->cache_default_attribute,
```

Regarding the returns definition, it should be an int :thinking:

Reply:
Yep, nice catch!